### DISCUSSION TITLE
Feature request: Use native events instead of jQuery events

### DISCUSSION THREAD

**Discussion by yjukaku** [activeadmin/activeadmin#7879] [General]
Thansk for your work on this gem! 

To start, I definitely understand that activeadmin is not actively looking to switch away from jQuery. That would be a massive effort. However, I think I've run into a particular problem that would justify moving away from jQuery _events_. 

In our app, we have a guideline in place to avoid jQuery unless necessary. We wanted to add some JS code to listen for the `has_many_remove:before` event, so we added:

```javascript
document.addEventListener("has_many_remove:before", (ev) => { doSomething() })
```

however, that's never triggered. This does get triggered, though:

```
$(document).on("has_many_remove:before", (ev) => { doSomething() })
```

The reason is that has_many.js is calling jQuery's .trigger function here: https://github.com/activeadmin/activeadmin/blob/8c4c589fcf100770e3375da5ccf69cec6de65ac8/app/javascript/active_admin/initializers/has-many.js#L20 . That function does _not_ cause native event listeners to fire.

If we were able to switch all of the `trigger` calls to `dispatchEvent`, we could have _both_ jQuery AND native event listeners to fire. However, because the `trigger` calls pass in arguments representing the item to remove and the parent element, I believe this would be a breaking change. Native events would have to store those elements in the event data's `detail` property, while jQuery passes those elements as arguments to the listener function:

```
document.addEventListener("has_many_remove:before", (ev) => { doSomething(ev.detail.toRemove, ev.detail.parent) })
// vs.
$(document).on("has_many_remove:before", (ev, toRemove, parent) => { doSomething(toRemove, parent) })
```

I'm not sure if there's a way to make this a non breaking change (probably not?).

Would AA be willing to make this breaking change?

**Comment by javierjulio** [comment]
This is unlikely. Perhaps if a new event could be introduced that doesn't use jQuery, I would consider that to help. I'll move this to discussions for now. Thank you.
